internal/compiler/diagnostics.rs

```diff
@@ -520,3 +520,85 @@ impl BuildDiagnostics {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_source_file_offset_line_column_mapping() {
+        let content = r#"import { LineEdit, Button, Slider, HorizontalBox, VerticalBox } from "std-widgets.slint";
```

Comment:
nitpick: this string is quite long and makes the test take a quite a lot of room in that file. 
It doesn't need to be so long for the test to work, i'd remove many elements.